src/sonic-yang-models/yang-templates/sonic-types.yang.j2

```diff
@@ -208,6 +208,14 @@ module sonic-types {
         }
     }
 
+    typedef mode {
```

Comment:
Can we use switchport_mode as type name?

Reply:
Yes we can. I think we may use it as "mode" type might be overlapping in some cases. I will modify it.

Reply:
updated as per suggestion